src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -101,6 +107,80 @@ public void setPersons(List<Person> persons) throws DuplicatePersonException {
         return FXCollections.unmodifiableObservableList(internalList);
     }
 
+    /**
+     * Sorts players by selected field in asc or desc order.
+     * @return
+     */
+    public void sortBy(String field, String order) throws NoPlayerException {
+        if (internalList.size() < 1) {
+            throw new NoPlayerException();
+        }
+
+        Comparator<Person> comparator = null;
+
+        Comparator<Person> nameComparator = new Comparator<Person>() {
+            @Override
+            public int compare(Person p1, Person p2) {
+                return p1.getName().fullName.compareTo(p2.getName().fullName);
+            }
+        };
+
+        Comparator<Person> phoneComparator = new Comparator<Person>() {
+            @Override
+            public int compare(Person p1, Person p2) {
+                return p1.getPhone().value.compareTo(p2.getPhone().value);
+            }
+        };
+
+        Comparator<Person> emailComparator = new Comparator<Person>() {
+            @Override
+            public int compare(Person p1, Person p2) {
+                return p1.getEmail().value.compareTo(p2.getEmail().value);
+            }
+        };
+
+        Comparator<Person> addressComparator = new Comparator<Person>() {
+            @Override
+            public int compare(Person p1, Person p2) {
+                return p1.getAddress().value.compareTo(p2.getAddress().value);
+            }
+        };
+
+        switch (field) {
```

Comment:
I think this part for switch formatting, Travis might throw another checkstyle error.

[Configure IntelliJ Checkstyle](https://github.com/CS2103JAN2018-F14-B1/main/blob/master/docs/UsingCheckstyle.adoc)